File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();

        for (int i=0; i<nums.length; i++) {
            numMap.put(nums[i], i);
        }

        for (int i=0; i<nums.length; i++) {
            int rest = target - nums[i];
            if (numMap.containsKey(rest) && numMap.get(rest) != i) {
                return new int[] {i, numMap.get(rest)};
            }
        }
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numMap = new Dictionary<int, int>();

        for (int i=0; i< nums.Length; i++){
            int rest = target - nums[i];
            if (numMap.ContainsKey(rest)){
                return new int[] {numMap[rest], i};
            }
            numMap[nums[i]] = i;
        }
        return new int[] {};
    }
}